Release ticloud cli v1.0.0-beta.2 (#18938)
pingcap/docs: tidb-cloud/ticloud-serverless-sql-user-update.md

---
title: ticloud serverless sql-user update
summary: The reference of `ticloud serverless sql-user update`.
---

# ticloud serverless sql-user update

> Update a TiDB Cloud Serverless SQL user

Reviewer: ```suggestion
Update a TiDB Cloud Serverless SQL user:
```

```shell
ticloud serverless sql-user update [flags]
```

## Examples

Update a TiDB Cloud Serverless SQL user in interactive mode:

```shell
ticloud serverless sql-user update
```

Update a TiDB Cloud Serverless SQL user in non-interactive mode:

```shell
ticloud serverless sql-user update -c <cluster-id> --user <user-name> --password <password> --role <role>
```

## Flags

In non-interactive mode, you need to manually enter the required flags. In interactive mode, you can just follow CLI prompts to fill them in.

| Flag                    | Description                                                                   | Required | Note                                                 |
|-------------------------|-------------------------------------------------------------------------------|----------|------------------------------------------------------|

> | -c, --cluster-id string | Specifies the ID of the cluster.                                              | Yes      | Only works in non-interactive mode.                  |

Reviewer: ```suggestion
| -c, --cluster-id string | The ID of the cluster.                                              | Yes      | Only works in non-interactive mode.                  |
```

| --password string       | The new password of the SQL user.                                             | No       | Only works in non-interactive mode.                  |

> | --role strings          | The new role(s) of the SQL user. Passing this flag replaces preexisting data. | No       | Only works in non-interactive mode.                  |

Reviewer: ```suggestion
| --role strings          | The new roles of the SQL user. Passing this flag replaces preexisting data. | No       | Only works in non-interactive mode.                  |
```

> | --add-role strings      | The role(s) to be added to the SQL user.                                      | No       | Only works in non-interactive mode.                  |

Reviewer: ```suggestion
| --add-role strings      | The roles to be added to the SQL user.                                      | No       | Only works in non-interactive mode.                  |
```

> | -c, --cluster-id string | The ID of the cluster.                                              | Yes      | Only works in non-interactive mode.                  |
> | --password string       | The new password of the SQL user.                                             | No       | Only works in non-interactive mode.                  |
> | --role strings          | The new roles of the SQL user. Passing this flag replaces preexisting data. | No       | Only works in non-interactive mode.                  |
> | --add-role strings      | The roles to be added to the SQL user.                                      | No       | Only works in non-interactive mode.                  |
> | --delete-role strings   | The roles to be deleted from the SQL user.                                  | No       | Only works in non-interactive mode.                  |
> | -u, --user string       | The name of the SQL user to be updated.                                       | No       | Only works in non-interactive mode.                  |

Reviewer 2: ```suggestion
| -c, --cluster-id string | Specifies the ID of the cluster.                                              | Yes      | Only works in non-interactive mode.                  |
| --password string       | Specifies the new password of the SQL user.                                             | No       | Only works in non-interactive mode.                  |
| --role strings          | Specifies the new roles of the SQL user. Passing this flag replaces existing roles. | No       | Only works in non-interactive mode.                  |
| --add-role strings      | Specifies the roles to be added to the SQL user.                                      | No       | Only works in non-interactive mode.                  |
| --delete-role strings   | Specifies the roles to be deleted from the SQL user.                                  | No       | Only works in non-interactive mode.                  |
| -u, --user string       | Specifies the name of the SQL user to be updated.                                       | No       | Only works in non-interactive mode.                  |
```
